src/main/java/io/github/dsheirer/audio/AudioUtils.java

```diff
@@ -78,4 +81,44 @@ public static byte[] convert(List<float[]> audioBuffers)
 
         return stream.toByteArray();
     }
+
+    /**
+     * Normalize audio samples
+     */
+    public static List<float[]> normalize(List<float[]> audioBuffers)
+    {
+        List<float[]> normalizedBuffers = new ArrayList<float[]>();
+        float targetMax = 0.95f; // -0.44dBFS
+        float max = 0.1f; // Initialize at 0.1 (-20dBFS) to limit maximum gain
+
+        for(float[] audioBuffer: audioBuffers)
+        {
+            for(float value: audioBuffer)
+            {
+                float newMax = Math.max(max, Math.abs(value));
+                if(!Float.isNaN(newMax))
+                {
+                    max = newMax;
+                }
+            }
+        }
+
+        if(targetMax == max)
+        {
+            return audioBuffers;
+        }
+
+        for(float[] audioBuffer: audioBuffers)
+        {
+            float[] newBuffer = new float[audioBuffer.length];
+            for(int i=0;i<audioBuffer.length;i++)
+            {
+                newBuffer[i] = audioBuffer[i] * targetMax / max;
```

Comment:
You might consider pre-calculating a gain value outside of the loops to minimize the workload:

```
float gain = targetMax / max;

for(float[] ...)
{
   for(i = ...)
   {
      audiobuffer[i] *= gain;
   }
{
```